lib/Ohmbrewer_Thermostat.cpp

```diff
@@ -97,58 +103,81 @@ Ohmbrewer::Thermostat::Thermostat(int id, int (&thermPins)[3], const double targ
 /**
  * Constructor
  * @param id The Sprout ID to use for this piece of Thermostat
- * @param tubePins[ temp busPin ; heating powerPin ; heating controlPin ]
+ * @param thermPins list with formatting of: [ temp busPin ;  heating controlPin ; heating powerPin ]
  * @param stopTime The time at which the Thermostat should shut off, assuming it isn't otherwise interrupted
  * @param state Whether the Thermostat is ON (or OFF). True => ON, False => OFF
  * @param currentTask The unique identifier of the task that the Thermostat believes it should be processing
  */
-Ohmbrewer::Thermostat::Thermostat(int id, int (&thermPins)[3], int stopTime,
+Ohmbrewer::Thermostat::Thermostat(int id, std::list<int>* thermPins, int stopTime,
                                   bool state, String currentTask) : Ohmbrewer::Equipment(id, stopTime, state, currentTask) {
-    int n = sizeof(thermPins) / sizeof(int);
-    if ( n > 2){
-        _heatingElm = new HeatingElement(id+2, thermPins[1], thermPins[2]);
-        _tempSensor = new TemperatureSensor(id+1, thermPins[0]);
-    }else if (n == 2 ){                     // no switch, only power% (SSR in parallel or always on other leg)
-        _heatingElm = new HeatingElement(id+2, thermPins[1], -1);
-        _tempSensor = new TemperatureSensor(id+1, thermPins[0]);
-    }else{
-        //publish error
+    //PID set up
```

Comment:
Ditto.

If you're doing the same thing in all of these, maybe you should make a ```initPID``` method and call it from the constructors?